proto/internal/temporal/server/api/persistence/v1/queues.proto

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2020 Temporal Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+syntax = "proto3";
+
+package temporal.server.api.persistence.v1;
+option go_package = "go.temporal.io/server/api/persistence/v1;persistence";
+
+import "temporal/server/api/persistence/v1/predicates.proto";
+
+message QueueAckLevel {
+    int64 ack_level = 1;
+    map<string, int64> cluster_ack_level = 2;
+}
+
+message QueueState {
+    map<int32, QueueReaderState> reader_states = 1;
+    int64 exclusive_max_read_key = 2;
```

Comment:
This is the boundary of read & write zone. Needed since the slice with the highest max key can be completed and removed before other slices. Or there can be no slices at all if the process for all pending slices are completed.

Reply:
should this be a struct (not int64, e.g. timer)
and can you find a better name

Reply:
I can define a separate struct for task key and persist both the fire time and the taskID (and use it in QueueSliceRange as well). 

But since the actual read/delete persistence api only supports querying either time or taskID, persisting one value seems sufficient to me. Let me know.

I tried.. naming is so hard, maybe readerHighWatermark? Any suggestions?

Reply:
Defined new `taskKey` message and used it here and in QueueSliceRange.

Renamed the field to `exclusive_reader_high_watermark`.